### PR TITLE
[ci:force] Migrate to NG15

### PR DESCRIPTION
**Pull request by VitoAlbano**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by wojd0 — APPROVED** [review]

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
         description: 'devel'
         required: false
         type: string
-        default: false
+        default: 'false'
```
Comment (DenysVuika) [inline .github/workflows/pull-request.yml:15]: is conversion to string needed here?

File: lib/process-services/karma.conf.js [hunk 2]
```diff
@@ -15,7 +16,7 @@ module.exports = function (config) {
                 included: true,
                 watched: false
             },
-            { pattern: 'node_modules/chart.js/dist/Chart.js', included: true, watched: false },
+            { pattern: 'node_modules/chart.js/dist/Chart.min.js', included: true, watched: false },
```
Comment (DenysVuika) [inline lib/process-services/karma.conf.js:19]: I don't see .min.js file in the node modules, is that a new version?

File: lib/extensions/src/lib/components/dynamic-column/dynamic-column.component.ts
Comment (DenysVuika) [inline lib/extensions/src/lib/components/dynamic-column/dynamic-column.component.ts:1]: incorrect formatting, the change should be discarded

File: lib/process-services/karma.conf.js (hunk 2 above)
Comment (VitoAlbano) [reply lib/process-services/karma.conf.js:19]: I followed the suggestion 'cause it was making it fail the whole task, but thinking again now, do we actually need this? i've run the unit test without and they work fine

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (VitoAlbano) [reply .github/workflows/pull-request.yml:15]: it was making the checker failing and as you see it's type string it make sense to have it a default value string

File: docs/release-notes/RelNote410.md
Comment (DenysVuika) [inline docs/release-notes/RelNote410.md:1]: the .md files should not be affected... is there a way to disable .md reformatting?

**Comment by DenysVuika** [comment]
@VitoAlbano we have a task in A/N BDU to cleanup ADF from the material styling where possible, so when we re-generate the migration, it will be less files modified.

File: docs/release-notes/RelNote410.md
Comment (VitoAlbano) [reply docs/release-notes/RelNote410.md:1]: I'll investigate that

**Comment by VitoAlbano** [comment]
@DenysVuika i'm fixing the e2es at the moment which has the same problem as the unit test, so i'll rebase this from time to time hopefully less changes will be there 🕺

File: demo-shell/src/app/components/file-view/file-view.component.html [hunk 3]
```diff
@@ -157,7 +157,7 @@ <h1>My custom toolbar</h1>
         </adf-info-drawer-tab>
 
         <adf-info-drawer-tab label="Versions">
-            <mat-card>
+            <mat-card appearance="outlined">
```
Comment (DenysVuika) [inline demo-shell/src/app/components/file-view/file-view.component.html:160]: this one will be removed before migration by https://github.com/Alfresco/alfresco-ng2-components/pull/9050

File: demo-shell/src/app/components/card-view/card-view.component.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <div class="app-main-content">
     <h1>CardView Component</h1>
 
-    <mat-card class="app-card-view">
+    <mat-card appearance="outlined" class="app-card-view">
```
Comment (DenysVuika) [inline demo-shell/src/app/components/card-view/card-view.component.html:4]: this one will be removed before migration by https://github.com/Alfresco/alfresco-ng2-components/pull/9050

File: demo-shell/src/app/components/files/files.component.scss [hunk 5]
```diff
@@ -7,7 +7,7 @@
         margin: 0;
     }
 
-    .app-show-versions-button.mat-icon-button {
+    .app-show-versions-button.mat-mdc-icon-button {
```
Comment (DenysVuika) [inline demo-shell/src/app/components/files/files.component.scss:10]: already removed on develop

File: demo-shell/src/app/components/login/login.component.scss [hunk 6]
```diff
@@ -1,4 +1,4 @@
-.app-setting-button.mat-fab {
+.app-setting-button.mat-mdc-fab {
```
Comment (DenysVuika) [inline demo-shell/src/app/components/login/login.component.scss:1]: will be cleaned in a separate PR coming soon

File: demo-shell/src/app/components/settings/host-settings.component.html [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <div class="adf-setting-container">
-    <mat-card class="adf-setting-card">
+    <mat-card appearance="outlined" class="adf-setting-card">
```
Comment (DenysVuika) [inline demo-shell/src/app/components/settings/host-settings.component.html:2]: already removed in develop

File: e2e/content-services/pages/permissions.page.ts
Comment (DenysVuika) [inline e2e/content-services/pages/permissions.page.ts:1]: this file was clened up from demo shell and e2e

File: e2e/core/pages/content-services.page.ts [hunk 8]
```diff
@@ -106,7 +106,7 @@ export class ContentServicesPage {
     }
 
     async clickHyperlinkNavigationToggle(): Promise<void> {
```
Comment (DenysVuika) [inline e2e/core/pages/content-services.page.ts:108]: removed from code on develop

File: e2e/core/pages/content-services.page.ts [hunk 9]
```diff
@@ -272,7 +272,7 @@ export class ContentServicesPage {
     }
 
     async enableInfiniteScrolling(): Promise<void> {
-        const infiniteScrollButton = element(by.cssContainingText('.mat-slide-toggle-content', 'Enable Infinite Scrolling'));
+        const infiniteScrollButton = element(by.cssContainingText('.mat-mdc-slide-toggle', 'Enable Infinite Scrolling'));
```
Comment (DenysVuika) [inline e2e/core/pages/content-services.page.ts:275]: removed from code on develop

File: migrations.json
Comment (DenysVuika) [inline migrations.json:1]: this file should be removed after migrations, and should not be in the repo
